src/app/core/services/label.service.ts

```diff
@@ -28,7 +28,7 @@ export class LabelService {
   simpleLabels: SimpleLabel[];
 
   private labelsPollSubscription: Subscription;
-  private labelsSubject = new BehaviorSubject<SimpleLabel[]>([]);
+  private labelsSubject = new Subject<SimpleLabel[]>();
```

Comment:
The reason for changing ``labelsSubject'' from type ''BehaviourSubject'' to ''Subject'' is because ''BehaviourSubject'' emits an initial empty array as the all labels set. 
As a result, when opening a repo through a url with some specified labels in the url, those labels are sanitized immediately because of the initial emission of the empty all labels.
The change to ''Subject'' removes this initial empty labels emission, solving the issue.